pkg/storage/s3.go

```diff
@@ -0,0 +1,48 @@
+package storage
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/backup"
+	"github.com/spf13/pflag"
+)
+
+const (
+	s3EndpointOption = "s3.endpoint"
+)
+
+// S3EndpointOption is a BackendOption for changing the endpoint of the storage.
```

Comment:
This is a little complicated, for each option, we have to define one struct and two functions, suggest making it easier to build options.